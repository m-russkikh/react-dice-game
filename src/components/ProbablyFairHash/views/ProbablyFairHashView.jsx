import React from 'react';
import { FormGroup } from 'reactstrap';

import { ProbablyFairHashViewPropType } from '../propTypes';


function ProbablyFairHashView({ hash }) {
  return (
    <FormGroup>
      Provably Fair Hash: <b>{hash}</b>
    </FormGroup>
  );
}

ProbablyFairHashView.propTypes = ProbablyFairHashViewPropType;

export default ProbablyFairHashView;
