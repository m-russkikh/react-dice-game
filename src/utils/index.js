import { MAX_BET_NUMBER } from '../constants/diceGame';


export function saveState(state) {
  const serializedBalance = JSON.stringify(state.balance);

  localStorage.setItem('balance', serializedBalance);
}

export function getBetNumber() {
  return Math.floor(Math.random(0, 1) * MAX_BET_NUMBER) + 1;
}
