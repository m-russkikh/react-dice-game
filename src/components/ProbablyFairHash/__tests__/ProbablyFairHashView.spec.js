import { getShallowWrapper } from '../../../utils/testUtils';
import ProbablyFairHashView from '../views/ProbablyFairHashView';


describe('ProbablyFairHashView', () => {
  let props;
  let wrapper;

  beforeAll(() => {
    props = {
      nexBetNumber: 50,
    };

    wrapper = getShallowWrapper(ProbablyFairHashView, props);
  });

  describe('render', () => {
    it('should display correct', () => {
      expect(wrapper.debug()).toMatchSnapshot();
    });
  });
});
