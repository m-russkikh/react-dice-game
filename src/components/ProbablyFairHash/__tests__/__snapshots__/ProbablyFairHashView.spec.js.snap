// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProbablyFairHashView render should display correct 1`] = `
"<FormGroup tag=\\"div\\">
  Provably Fair Hash: 
  <b>
    c0c7c76d30bd3dcaefc96f40275bdc0a
  </b>
</FormGroup>"
`;
